### ISSUE TITLE
Undefined/unused variables

### ISSUE THREAD

**Issue by gobiwan** [hsauers5/OptionsScraper#1]
Hello,

I'm a complete newbie trying to learn Python, and really want to be able to make this options scraper work. However, when I try to run the main.py, two errors are generated:

- The i variable for your range(0,7) loop in the "Vet timestamp" section is unused. Is that correct? I always get the "Check your connection. Market is closed" statements, when I'm actually running this when the market is open--is that because the failure of the "for" loop with the i variable isn't working, and therefore I'm not getting appropriate data returned?

Also, a smaller issue I ran into is that you have a print command that is calling an undefined variable ("options_stamp"). See line 26:
 # print(datetime.datetime.fromtimestamp(options_stamp))

Was the intent to print options_day?

Can anyone help--again, I'm very new to this and am struggling to understand how to get this script to work.

**Comment by hsauers5** [comment]
Hi, 

Thanks for reaching out. It's been a bit since I've maintained this project, so feel free to try out the corresponding tutorial here: 
[](https://medium.com/free-code-camp/how-i-get-options-data-for-free-fba22d395cc8)

You are correct that the _i_ variable in that loop is unused. It must iterate seven times for the purpose of determining which day (variable "_dd_") is the most recent. 

You are also correct in that "_print(datetime.datetime.fromtimestamp(options_stamp))_" is unused -- it is commented out. This is an old variable that I had in originally, likely changed the name of or eliminated altogether, and missed the comment. 

I cloned the project and there does seem to be an issue pulling down options pricing. Note the following code: 
  _# if no options are returned - market must be closed. 
  if options == {}:
    print("Market is closed... Rescheduling in 24 hours.")
  else:
    # write to file
    write_to_csv(options)_

There is some exception being thrown, possibly due to Yahoo Finance changing the format. Feel free to debug this and submit a pull request, and I'll have a look when I get the chance.

**Comment by gobiwan** [comment]
Thanks for the quick reply.

As I mentioned, I'm a complete noob, so I'm struggling to figure out how to interpret some of your comments. 

Before I ask more of your time for clarification, you mentioned above that there's a corresponding tutorial for this? I'm not seeing a link or URL to it; could you provide that? I'll try digging through that to better understand the script before circling back to ask more questions.

Again, thanks for helping a rote beginner.

**Comment by hsauers5** [comment]
Good catch, I'm not sure how that got deleted. See here: https://www.freecodecamp.org/news/how-i-get-options-data-for-free-fba22d395cc8/

**Comment by gobiwan** [comment]
Thx. I'll dig through that first.
